crates/starknet-devnet-server/src/builder.rs

```diff
@@ -103,6 +110,52 @@ impl<TJsonRpcHandler: RpcHandler, THttpApiHandler: Clone + Send + Sync + 'static
                     .allow_methods(vec![Method::GET, Method::POST]),
             );
 
+        if config.log_request {
+            svc = svc.layer(axum::middleware::from_fn(request_logging_middleware));
+        }
+
         Ok(Server::try_bind(&self.address)?.serve(svc.into_make_service()))
     }
 }
+
+async fn request_logging_middleware(
+    request: Request<hyper::Body>,
+    next: Next<hyper::Body>,
+) -> Result<impl IntoResponse, Response> {
```

Comment:
Why is the Error enumeration type defined to be `Response`?

Reply:
[7a289c0](https://github.com/0xSpaceShard/starknet-devnet-rs/pull/455/commits/7a289c060224b8abbfa53bdd569411178a5a03c7)

Basically its just the same, it will be converted to Response, because (StatusCode, String) implements IntoResponse trait